src/custom/state/enhancedTransactions/updater/FinalizeTxUpdater.tsx

```diff
@@ -2,20 +2,23 @@
  * This file is basically a Mod of src/state/transactions/updater
  */
 
-import { useCallback, useEffect, useMemo } from 'react'
-import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { useEffect, useMemo } from 'react'
+import { useAppDispatch } from 'state/hooks'
 // import { SupportedChainId } from 'constants/chains'
 import { useActiveWeb3React } from 'hooks/web3'
-import { retry, RetryableError, RetryOptions } from 'utils/retry'
 import { updateBlockNumber } from 'state/application/actions'
 import { useAddPopup, useBlockNumber } from 'state/application/hooks'
 import { checkedTransaction, finalizeTransaction } from '../actions'
+import { EnhancedTransactionDetails, HashType } from '../reducer'
+import { GetReceipt, useGetReceipt } from 'hooks/useGetReceipt'
+import { useAllTransactionsDetails } from 'state/enhancedTransactions/hooks'
+import { Dispatch } from 'redux'
```

Comment:
we have useAppDispatch

Reply:
Yep, it's being used. It's just that some aux function receives the dispatch as a parameter. See `finalizeEthereumTransaction`